src/Proto3/Suite/DotProto/Parsing.hs

```diff
@@ -19,12 +22,25 @@ import Text.Parser.Char
 import Text.Parser.Combinators
 import Text.Parser.LookAhead
 import Text.Parser.Token
+import qualified Turtle
 
 ----------------------------------------
--- interface
-
-parseProto :: String -> Either ParseError DotProto
-parseProto = parse topLevel "" . stripComments
+-- interfaces
+
+-- | @parseProto mp inp@ attempts to parse @inp@ as a 'DotProto'. @mp@ is the
+-- module path to be injected into the AST as part of 'DotProtoMeta' metadata on
+-- a successful parse.
+parseProto :: Path -> String -> Either ParseError DotProto
+parseProto modulePath = parse (topLevel modulePath) "" . stripComments
```

Comment:
What is the reason for feeding the parsec parser a `String` instead of a `Text`? (other than `stripComments` returning a `String`...)